From fc215fed8bca43c8ad0069445077433994541ec7 Mon Sep 17 00:00:00 2001
From: Guy Nisan <gnisan@habana.ai>
Date: Mon, 22 Feb 2021 08:38:04 +0200
Subject: [PATCH] [SW-33162] gaudi: sp sram wipe only on cold boot

In preparation for boot info communication between u-boot/linux and
preboot, preboot may wipe SP SRAM only on cold boot. This wipe is needed
for initializing ECC, hence it's only relevant on cold boot.

Change-Id: I6ae3122ac4c8a122109c6bd44d4a85195e9b3583
Signed-off-by: Guy Nisan <gnisan@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index c8e5007..35ebcbc 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -345,10 +345,13 @@
 {
 	register int i;
 	u32 *data = (u32 *)(HL_SP_SRAM_BASE);
+	unsigned int rst_src = regmap_readl(mmPSOC_GLOBAL_CONF_RST_SRC);
 
-	/* SP SRAM access is limited to 32bit */
-	for (i = 0; i < HL_SP_SRAM_SIZE / sizeof(u32); i++)
-		data[i] = 0x5a5a5a5a;
+	if (rst_src == HL_COLD_RST) {
+		/* SP SRAM access is limited to 32bit */
+		for (i = 0; i < HL_SP_SRAM_SIZE / sizeof(u32); i++)
+			data[i] = 0x5a5a5a5a;
+	}
 
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SP_SRAM_EN, true);
 }
